app/coffee/BucketController.coffee

```diff
@@ -0,0 +1,30 @@
+settings = require("settings-sharelatex")
+logger = require("logger-sharelatex")
+FileHandler = require("./FileHandler")
+metrics = require("metrics-sharelatex")
+Errors = require('./Errors')
+
+module.exports = BucketController =
+
+	getFile: (req, res)->
+		{bucket} = req.params
+		key = req.params[0]
+		{format, style} = req.query
+		credentials = settings.filestore.s3[bucket]
```

Comment:
It is worth checking that the credentials exist or an error will be thrown further down the line.